### PR TITLE
[Layout foundations] Rebuild Page component with primitives

### LINKED ISSUE #8575: [Layout foundations] Align Page with SkeletonPage
SkeletonPage was rebuilt with layout primitives and adjusted to make sure it matches the Page component for spacing and alignment. This means that mixins that were previously shared between the two components are now missing and the two components can get our of sync with eachother

```[tasklist]
### Tasks
- [ ] Rebuild Page with layout primitives
- [ ] Remove all shared mixins
- [ ] Render Page component from SkeletonPage (no duplication of markup)
```

### PR DESCRIPTION
**Pull request by kyledurand**
### WHY are these changes introduced?

Fixes https://github.com/Shopify/polaris/issues/8575

### How to 🎩

[Spin link](https://admin.web.page-rebuild.kyle-durand.us.spin.dev/store/shop1)

### 🎩 checklist

- [ ] Tested on [mobile](https://github.com/Shopify/polaris/blob/main/documentation/Tophatting.md#cross-browser-testing)
- [ ] Tested on [multiple browsers](https://help.shopify.com/en/manual/shopify-admin/supported-browsers)
- [ ] Tested for [accessibility](https://github.com/Shopify/polaris/blob/main/documentation/Accessibility%20testing.md)
- [ ] Updated the component's `README.md` with documentation changes
- [ ] [Tophatted documentation](https://github.com/Shopify/polaris/blob/main/documentation/Tophatting%20documentation.md) changes in the style guide

### CODE REVIEW

File: polaris-react/src/components/Page/Page.tsx [hunk 1]
```diff
@@ -46,17 +43,32 @@ export function Page({
...
-
+  let pageWidth = '62.375rem';
+  if (fullWidth) pageWidth = '100%';
+  if (narrowWidth) pageWidth = '41.375rem';
   return (
-    <div className={pageClassName}>
-      {headerMarkup}
-      <div className={contentClassName}>{children}</div>
-    </div>
+    <Inline align="center">
```
Comment (chloerice) [inline polaris-react/src/components/Page/Page.tsx]: What does wrapping this in Inline do for the inner Box?

File: polaris-react/src/components/SkeletonPage/SkeletonPage.tsx [hunk 2]
```diff
@@ -30,85 +29,39 @@ export function SkeletonPage({
...
-              {primaryActionMarkup}
-            </Inline>
-          </Box>
-          <Box paddingBlockEnd="2" width="100%">
-            {children}
-          </Box>
-        </AlphaStack>
-      </Box>
-    </AlphaStack>
+    <Page
```
Comment (chloerice) [inline polaris-react/src/components/SkeletonPage/SkeletonPage.tsx:54]: This is genius 🙌🏽

**Review by chloerice — APPROVED** [review]
Tophat looks great 👍🏽 I went through each of the UI Tests and it looks like the difference is just due to fullWidth (and for whatever reason the level of zoom on the snapshot?) Nothing unexpected there layout-wise. Double checked full width pages to be sure 💯

File: polaris-react/src/components/Page/Page.tsx (hunk 1 above)
Comment (kyledurand) [reply polaris-react/src/components/Page/Page.tsx]: This is now centering the page content vs having `margin: 0 auto` previously

**Comment by laurkim** [comment]
> Tophat looks great 👍🏽 I went through each of the UI Tests and it looks like the difference is just due to fullWidth (and for whatever reason the level of zoom on the snapshot?) Nothing unexpected there layout-wise. Double checked full width pages to be sure 💯

I'm seeing this too in the storybook diff where the baseline is thinner, but the UI looks as expected in the Spin instance. I'm not sure if this is a result of chromatic acting up.

Also, the SkeletonPage with Narrow Width story wasn't updated in this PR but it looks like there's extra spacing being added between the SkeletonDisplayText.
21-01-279oy-elr69

**Comment by kyledurand** [comment]
@laurkim I noticed that as well and I can look into it. I'm wondering if that's an issue stemming from how old this branch is. because Card shadows etc are changing as well and this doesn't touch any internals of skeleton display text

**Comment by laurkim** [comment]
> @laurkim I noticed that as well and I can look into it. I'm wondering if that's an issue stemming from how old this branch is. because Card shadows etc are changing as well and this doesn't touch any internals of skeleton display text

@kyledurand Oh, good point, might be worth a rebase/merge off `main`. That might resolve some of the chromatic diff 🤞

**Pull request closed without merging**